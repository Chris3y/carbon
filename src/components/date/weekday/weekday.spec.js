import React from 'react';
import TestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import classicTheme from '../../../style/themes/classic';
import Weekday from './weekday.component';

describe('Weekday', () => {
  const shallowRender = (props, renderer = shallow) => {
    return renderer(<Weekday { ...props }>sample children</Weekday>);
  };

  it('renders presentational div and context provider for its children', () => {
    expect(shallowRender({ title: 'tile', className: 'class-name' }, TestRenderer.create)).toMatchSnapshot();
  });

  describe('classic theme', () => {
    it('applies custom styling', () => {
      expect(shallowRender({ theme: classicTheme }, TestRenderer.create)).toMatchSnapshot();
    });
  });
});
