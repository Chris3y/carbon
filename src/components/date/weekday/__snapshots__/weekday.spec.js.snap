// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weekday classic theme applies custom styling 1`] = `
<div
  className="sc-bdVaJa fObhsM"
  role="columnheader"
>
  <abbr
    className="sc-bwzfXH gdAzKI"
  >
    sample children
  </abbr>
</div>
`;

exports[`Weekday renders presentational div and context provider for its children 1`] = `
<div
  className="class-name sc-bdVaJa cwXVru"
  role="columnheader"
>
  <abbr
    className="sc-bwzfXH gdAzKI"
    title="tile"
  >
    sample children
  </abbr>
</div>
`;
